### PR TITLE
Make running of the generated tests optional

### LINKED ISSUE #65: Add an option to not run tests
Hello,

Thanks for this project!

Right now, `elm-verify-examples` compiles examples as tests, and then run them. If examples are compiled in the `tests/` directory, then it's likely they will be run by `elm-test` “manually” in another pass. Indeed, I think it's expected to see the user doing this:

```bash
$ elm-verify-examples --no-test-run … # Compile examples to tests
$ elm-test # Run all tests
```

Thoughts?

### PR DESCRIPTION
**Pull request by tad-lispy**
Closes #65

By default only generate the tests and leave them in the output directory. 

Add `--run` (`-r`) option to also run and then delete the files (i.e. work as before).

The ReadMe, tests and Changelog are updated accordingly.

### CODE REVIEW

File: bin/cli.js [hunk 1]
```diff
@@ -16,6 +16,9 @@ var argv = require('yargs')
   .alias("o", "output")
   .describe("output", "Change path to the generated tests.")
   .default("output", "tests")
+  .alias("r", "run")
```
Comment (stoeffel) [inline bin/cli.js]: I think I'd call it `run-tests`, to be more explicit.

File: bin/cli.js [hunk 2]
```diff
@@ -16,6 +16,9 @@ var argv = require('yargs')
...
+  .describe("run", "Run generated tests.")
```
Comment (stoeffel) [inline bin/cli.js]:
```suggestion
  .describe("run", "Run generated tests using `elm-test`.")
```

File: bin/cli.js [hunk 3]
```diff
@@ -31,9 +34,11 @@ var model = runner.init(argv);
...
-  var status = runner.runElmTest(model);
-  if (status === 0) runner.cleanup(model);
-  runner.warnSummary(model, warnings);
-  process.exit(status);
+  if (model.run) {
+    var status = runner.runElmTest(model);
+    if (status === 0) runner.cleanup(model);
+    runner.warnSummary(model, warnings);
+    process.exit(status);
+  }
```
Comment (stoeffel) [inline bin/cli.js]:
```suggestion
  } else {
    process.exit(0);
  }
```

File: bin/cli.js [hunk 4]
```diff
@@ -31,9 +34,11 @@ var model = runner.init(argv);
...
 runner.run(model, function(warnings) {
   warnings.map(runner.warnModule(model));
-  var status = runner.runElmTest(model);
-  if (status === 0) runner.cleanup(model);
-  runner.warnSummary(model, warnings);
-  process.exit(status);
+  if (model.run) {
+    var status = runner.runElmTest(model);
+    if (status === 0) runner.cleanup(model);
+    runner.warnSummary(model, warnings);
```
Comment (stoeffel) [inline bin/cli.js]: I'm not sure if I remember correctly, but I think we want this in either case, right?

File: package-lock.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
```
Comment (stoeffel) [inline package-lock.json]: Could you rebase on master and update this file accordingly, I've updated the dependencies on master? Thanks

File: bin/cli.js (hunk 4 above)
Comment (tad-lispy) [reply bin/cli.js]: Yeah, I think so.

File: package-lock.json (hunk 5 above)
Comment (tad-lispy) [reply package-lock.json]: Done.

Comment (tad-lispy) [reply package-lock.json]: Done.

**Comment by tad-lispy** [comment]
Thank you for taking time to review my changes. I believe I've addressed all your concerns. Please review again.

**Comment by stoeffel** [comment]
Will release asap.

**Comment by tad-lispy** [comment]
Hi! I hope you are doing well. A friendly reminder to publish new version on npmjs.com.

**Comment by stoeffel** [comment]
Done! Thanks for the reminder!

**Pull request merged**